core/src/main/scripts/importer/validateData.py

```diff
@@ -2964,129 +2963,82 @@ def checkLine(self, data):
+        'Site1_Ensembl_Transcript_Id', 
+        'Site2_Hugo_Symbol',
+        'Site2_Entrez_Gene_Id',
+        'Site2_Chromosome',
+        'Site2_Position',
+        'Site2_Region_Number',
+        'Site2_Ensembl_Transcript_Id']
+
+    # TODO: Revive the support to validate the events based on transcripts and exons - once the new visualization features (UI for fusion tab, breakpoint visualization) are added to front-end.
+    
+    def __init__(self, *args, **kwargs):
         super(StructuralVariantValidator, self).__init__(*args, **kwargs)
-        self.structural_variant_entries = {}
+        self.sv_entries = {}
         self.ncbi_build = None
-        self.transcript_set = set()
-        self.transcript_exons_dict = {}
+        self.unique_fields = []
+        self.other_required_genomic_fields = ['Site1_Region', 'Site2_Region', 'Site1_Contig', 'Site2_Contig']
+        self.missing_genomic_fields = []
 
     def checkHeader(self, data):
         num_errors = super(StructuralVariantValidator, self).checkHeader(data)
 
         # Check presence of gene
         if not ('Site1_Hugo_Symbol' in self.cols or 'Site1_Entrez_Gene_Id' in self.cols):
-            self.logger.error('Structural variant requires Site1_Entrez_Gene_Id and/or Site1_Hugo_Symbol column',
+            self.logger.error('Structural variant event requires Site1_Entrez_Gene_Id and/or Site1_Hugo_Symbol column',
                               extra={'line_number': self.line_number})
             num_errors += 1
 
         # Check second gene column
         if not ('Site2_Hugo_Symbol' in self.cols or 'Site2_Entrez_Gene_Id' in self.cols):
-            self.logger.error('Fusion event requires Site2_Entrez_Gene_Id and/or Site2_Hugo_Symbol column',
+            self.logger.error('Structural variant event requires Site2_Entrez_Gene_Id and/or Site2_Hugo_Symbol column',
                               extra={'line_number': self.line_number})
             num_errors += 1
 
-        # Fusion events should be described by exons.
-        # Using chromosomal locations to describe fusion events is not supported.
-        if not {'Site1_Exon', 'Site2_Exon'}.issubset(set(self.cols)):
-            self.logger.error('Fusion event requires "Site1_Exon" and "Site2_Exon" columns',
-                              extra={'line_number': self.line_number})
-            num_errors += 1
-
-        if not {'Site1_Ensembl_Transcript_Id', 'Site2_Ensembl_Transcript_Id'}.issubset(set(self.cols)):
-            self.logger.error('Fusion event requires "Site1_Ensembl_Transcript_Id" and "Site2_Ensembl_Transcript_Id" '
-                              'columns',
-                              extra={'line_number': self.line_number})
-            num_errors += 1
+        # Check the columns to define the uniqueness of SV records
+        # Report the missing genomic fields
+        for value in self.UNIQUENESS_COLUMNS:
+            if value in self.cols:
+                self.unique_fields.append(value)
```

Comment:
I'm assuming this unique_fields is used later somewhere else? Otherwise is there a reason not to just combine other required genomic fields with the uniqueness columns?

Reply:
other required genomic fields has a list of free text genomic cols. they are not used in defining the uniqueness of the SV records but are still recommend.